packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -1128,6 +1276,51 @@ export class Table extends TableBase {
   protected get hasIndex(): boolean {
     return this.globalSecondaryIndexes.length + this.localSecondaryIndexes.length > 0;
   }
+
+  /**
+   * Set up key properties and return the Table encryption property from the
+   * user's configuration.
+   */
+  private parseEncryption(props: TableProps): {sseSpecification: CfnTable.SSESpecificationProperty, encryptionKey?: kms.IKey} {
+    let encryptionType = props.encryption;
+
+    if (encryptionType !== undefined && props.serverSideEncryption) {
```

Comment:
You want to also catch when `props.serverSideEncryption` was (explicitly) set to `false`, and `encryption` was provided.

```suggestion
    if (encryptionType != null && props.serverSideEncryption != null) {
```